### DISCUSSION TITLE
Request: do not remove `argType.defaultValue` in v 7.0

### DISCUSSION THREAD

**Discussion by MichaelAllenWarner** [storybookjs/storybook#18401] [Ideas]
I understand that `argType.defaultValue` has been deprecated, as explained [here](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#no-longer-inferring-default-values-of-args), where it's recommended: "If you were using `argType.defaultValue` or relying on inference to set a default value for an arg, you should now set a value for the arg at the component level."

My 2¢ here is that using `argType.defaultValue`, co-located as it is with the rest of a story's arg-related information, is more convenient than having to put a default arg-value in the component-level metadata object (and results in more readable code for the same reason). It's also more flexible, since you can give each story in the file its _own_ default value for a given arg.

What's great about this feature is that it gives you the option of setting a default arg-value (that's in effect when the user visits the story in the UI) _without_ having to set it on the story's actual `.args` object. I don't use it often, but sometimes I just don't want that value on the story's `.args`, and when that happens I'm glad the option is there.

**Comment by shilman** [comment]
@MichaelAllenWarner to make sure we're on the same page...

### Before

```js
export default { 
  argTypes: {
    foo: { defaultValue: 4 }
  },
};
```

### After

```js
export default {
  args: {
    foo: 4,
  }
}
```

I'm confused why the latter could be considered less readable than the former.

**Comment by MichaelAllenWarner** [comment]
@shilman 

Ah, I see the confusion. I'm talking about using `defaultValue` in the _story's_ `.argTypes`, like this (CSF 3.0):

```javascript
export const Primary = {
  // ...
  argTypes: {
    // ...
    foo: {
      // ...
      defaultValue: 4
    }
  }
}

export const Secondary = {
  ...Primary,
  argTypes: {
    ...Primary.argTypes,
    foo: {
      ...Primary.argTypes.foo,
      defaultValue: 8
    }
  }
}
```

The deprecation advice would instead have me set `args.foo` on the default-export (as you have in your "After"). I'd find that less convenient, and I _think_ it would prevent me from doing the story-specific `defaultValue` as I've done with Secondary there.
